### ISSUE TITLE
Unresponsive 0.0.2

### ISSUE THREAD

**Issue by delaneyj** [jscad/jscad-desktop#26]
[Have a project](https://github.com/delaneyj/cubeCore) that has not changed over the last 3 days.  After pulling latest (0.0.2) I get a `processing please wait` indefinitely.  My design isn't that complicated yet and on a Nvidia GTX1080.  Not sure what changed but maybe timeouts or reducing the default polys in the UI is necessary.

**Comment by kaosat-dev** [comment]
Hi again @delaneyj !
Cool project ! :) 
Normally I have not touched the way designs are processed , do you see anything in the console (you can switch it on with the same keyboard shortcut as in chrome)

**Comment by kaosat-dev** [comment]
I will take your project for a spin after work to see if there is anything obvious that I can find

**Comment by delaneyj** [comment]
Yeah I've been extensively using the debugger.  If you change the package.json to `./components/corexy.js` you can see it working,  remove from corexy from main.js and it works as well.  But the combo together doesn't work nor gives any errors.

**Comment by delaneyj** [comment]
I updated my example now corexy fails at well.  Interestingly I have a debugger call right at the beginning that doesn't get hit (or I'm missing because dev tools isn't the default and may miss the window).  Can't tell if its a jsdesktop or core openjscad issue.  I also tried changing the default resolution with no effect.

**Comment by kaosat-dev** [comment]
Hi again @delaneyj !
Sorry for the long delay, was sick for a while.
So , please checkout the latests version, there are lots of fixes and improvements ! 
- background processing of solids being the major one, so no more app crash if something goes wrong
- parameter /state handling should be a bit better (but not all bugs are solved in there)
I dug through your design and found the issue : 
- `motorBackOutsideBelt` is defined with a cube that has one dimensions of size 0 
- for now there are no errors thrown for a lot of invalid parameters in jscad/csg but that will change soonish, I hate it :D

**Comment by kaosat-dev** [comment]
<img width="1058" alt="screen shot 2018-01-31 at 13 27 38" src="https://user-images.githubusercontent.com/623281/35623126-8b7b8094-068a-11e8-91c5-9c47b8497c23.png">

**Comment by kaosat-dev** [comment]
There is still something breaking when specifying those dimensions in main() for coreXY btw, it has me stumped : I narrowed it down to let belt = union() failing silently ?

**Comment by kaosat-dev** [comment]
@delaneyj sorry but found the solution : pass the inputs to union as an array
```javascriptlet 
belt = union([
    rightYBelt,
    leftYBelt,

    motorInsideBelt,
    motorOutsideBelt,
    motorBackOutsideBelt
  ])
```

and its MUCH faster than without it :

**Comment by delaneyj** [comment]
Way better!  I had it working enough to make more changes, but back to weird state.  Wish a knew more to know how to debug this.  pull my project to see changes.
